### PR TITLE
#3283 Validate not required field 

### PR DESCRIPTION
**Pull request by ElementalArchmage**
support not required fields. Add checkbox Required in Rule. If true, check the pattern always. If false check the pattern, only if this one is present in the request

### CODE REVIEW

**Comment by brasseld** [comment]
Great work @ElementalArchmage 
Will have a look asap!

File: pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
     <groupId>io.gravitee.policy</groupId>
     <artifactId>gravitee-policy-request-validation</artifactId>
-    <version>1.6.0-SNAPSHOT</version>
+    <version>1.6.1-SNAPSHOT</version>
```
Comment (brasseld) [inline pom.xml]: No reason fore a 1.6.1-SNAPSHOT.

Please keep 1.6.0-SNAPSHOT

File: src/main/java/io/gravitee/policy/requestvalidation/RequestValidationPolicy.java [hunk 2]
```diff
@@ -140,12 +141,16 @@ public void end() {
...
-        for(Rule rule : configuration.getRules()) {
-            Validator validator = new ExpressionBasedValidator(executionContext.getTemplateEngine());
-            ConstraintViolation constraintViolation = validator.validate(rule.getInput(), rule.getConstraint());
-            if (constraintViolation != null) {
-                violations.add(constraintViolation);
+        for (Rule rule : configuration.getRules()) {
+
+            TemplateEngine templateEngine = executionContext.getTemplateEngine();
+            String input = templateEngine.getValue(rule.getInput(), String.class);
+            if (rule.getIsRequired() || (input != null && !input.equals(""))) {
```
Comment (brasseld) [inline src/main/java/io/gravitee/policy/requestvalidation/RequestValidationPolicy.java]: `!input.equals("")` => `!input.isEmpty()`

File: src/main/resources/schemas/schema-form.json [hunk 3]
```diff
@@ -129,6 +129,12 @@
             "title": "Field value",
             "description": "The input value to validate (supports EL)"
           },
+          "isRequired" : {
+            "type" : "boolean",
+            "title": "Required",
```
Comment (brasseld) [inline src/main/resources/schemas/schema-form.json]: `Required` => `Apply if value present`

Comment (ElementalArchmage) [reply src/main/resources/schemas/schema-form.json]: In both cases pattern apply, if value present. So this title won't give informathion about how it works. Maybe, will I rename title  "Require that value present" or something like that?

Comment (brasseld) [reply src/main/resources/schemas/schema-form.json]: Just leave it that way, that's ok.

**Comment by brasseld** [comment]
Hey @ElementalArchmage 

Can you squash the two commit into a single one (to keep a good history).

Once done, I'll merge.

Regards,

**Comment by ElementalArchmage** [comment]
> Hey @ElementalArchmage
> 
> Can you squash the two commit into a single one (to keep a good history).
> 
> Once done, I'll merge.
> 
> Regards,

I done it.

**Comment by brasseld** [comment]
Thanks a lot @ElementalArchmage !

**Pull request merged**